server/plugin.go

```diff
@@ -204,6 +208,91 @@ func (p *Plugin) OnActivate() error {
 	}
 
 	go p.initStats()
+
+	go func() {
+		time.Sleep(time.Second * 10)
+
+		instances, err := p.instanceStore.LoadKnownJIRAInstances()
+		if err != nil {
+			p.API.LogError("unable to register autolinks", "err", err)
+			return
+		}
+
+		for url := range instances {
+			instance, err := p.instanceStore.LoadJIRAInstance(url)
+			if err != nil {
+				continue
+			}
+
+			jci, ok := instance.(*jiraCloudInstance)
+			if !ok {
+				p.API.LogWarn("only cloud instances supported for autolink", "err", err)
+				continue
+			}
+
+			if err := p.InstallAutolinkForCloudInstance(jci); err != nil {
+				p.API.LogWarn("could not install autolinks for cloud instance", "instance", jci.BaseURL, "err", err)
+				continue
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (p *Plugin) InstallAutolinkForCloudInstance(jci *jiraCloudInstance) error {
+	client, err := jci.getJIRAClientForServer()
+	if err != nil {
+		return fmt.Errorf("unable to get jira client for server: %w", err)
+	}
+
+	keys, err := JiraClient{Jira: client}.GetAllProjectKeys()
+	if err != nil {
+		return fmt.Errorf("unable to make jira client: %w", err)
+	}
+
+	for _, key := range keys {
+		err = p.InstallAutolink(key, jci.BaseURL)
+	}
+	if err != nil {
+		return fmt.Errorf("some keys where not installed: %w", err)
+	}
+
+	return nil
+}
+
+func (p *Plugin) InstallAutolink(key, baseURL string) error {
+	baseURL = strings.TrimRight(baseURL, "/")
+	installList := []link.Link{
+		{
+			Name:     key + " key to link for " + baseURL,
+			Pattern:  `(` + key + `)(-)(?P<jira_id>\d+)`,
+			Template: `[` + key + `-${jira_id}](` + baseURL + `/browse/` + key + `-${jira_id})`,
+		},
+		{
+			Name:     key + " link to key for " + baseURL,
+			Pattern:  `(` + strings.ReplaceAll(baseURL, ".", `\.`) + `/browse/)(` + key + `)(-)(?P<jira_id>\d+)`,
+			Template: `[` + key + `-${jira_id}](` + baseURL + `/browse/` + key + `-${jira_id})`,
+		},
+	}
+
+	for _, toInstall := range installList {
+		linkBytes, err := json.Marshal(&toInstall)
+		if err != nil {
+			return err
+		}
+
+		req, err := http.NewRequest("POST", "/mattermost-autolink/api/v1/link", bytes.NewReader(linkBytes))
```

Comment:
Is this _really_ the current autolink's plugin ID? (I'm sure it is). The wild west...

0/5 maybe const it at the top, to highlight the dependency?

Reply:
Uggg. Ya. I originally put `autolink` and wondered why it didn't work. 
Ya a constant makes sense.